aiohue/sensors.py

```diff
@@ -53,3 +86,367 @@ def state(self):
+
+    @property
+    def presence(self):
+        return self._presence
+
+    async def set_config(self, on=None):
+        """Change config of a CLIP Presence sensor."""
+        data = {
+            key: value for key, value in {
+                'on': on,
+            }.items() if value is not None
+        }
+
+        await self._request('put', 'sensors/{}/config'.format(self.id),
+                            json=data)
+
+
+class ZLLPresenceSensor(GenericZLLSensor):
+    def __init__(self, id, raw, request):
+        super().__init__(id, raw, request)
+        self._presence = raw['state'].get('presence')
+
+    @property
+    def presence(self):
+        return self._presence
+
+    async def set_config(self, on=None, sensitivity=None, sensitivitymax=None):
+        """Change config of a ZLL Presence sensor."""
+        data = {
+            key: value for key, value in {
+                'on': on,
+                'sensitivity': sensitivity,
+                'sensitivitymax': sensitivitymax,
+            }.items() if value is not None
+        }
+
+        await self._request('put', 'sensors/{}/config'.format(self.id),
+                            json=data)
+
+
+class CLIPSwitchSensor(GenericCLIPSensor):
+    def __init__(self, id, raw, request):
+        super().__init__(id, raw, request)
+        self._buttonevent = raw['state'].get('buttonevent')
+
+    def buttonevent(self):
+        return self._buttonevent
+
+    async def set_config(self, on=None):
+        """Change config of a CLIP Switch sensor."""
+        data = {
+            key: value for key, value in {
+                'on': on,
+            }.items() if value is not None
+        }
+
+        await self._request('put', 'sensors/{}/config'.format(self.id),
+                            json=data)
+
+
+class ZGPSwitchSensor(GenericZGPSensor):
+    def __init__(self, id, raw, request):
+        super().__init__(id, raw, request)
+        self._buttonevent = raw['state'].get('buttonevent')
+
+    @property
+    def buttonevent(self):
+        if self._buttonevent == ZGP_SWITCH_BUTTON_1:
+            return ZGP_SWITCH_BUTTON_1
+        elif self._buttonevent == ZGP_SWITCH_BUTTON_2:
+            return ZGP_SWITCH_BUTTON_2
+        elif self._buttonevent == ZGP_SWITCH_BUTTON_3:
+            return ZGP_SWITCH_BUTTON_3
+        elif self._buttonevent == ZGP_SWITCH_BUTTON_4:
+            return ZGP_SWITCH_BUTTON_4
+
+    async def set_config(self, on=None):
+        """Change config of a ZGP Switch sensor."""
+        data = {
+            key: value for key, value in {
+                'on': on,
+            }.items() if value is not None
+        }
+
+        await self._request('put', 'sensors/{}/config'.format(self.id),
+                            json=data)
+
+
+class ZLLSwitchSensor(GenericZLLSensor):
+    def __init__(self, id, raw, request):
+        super().__init__(id, raw, request)
+        self._buttonevent = raw['state'].get('buttonevent')
```

Comment:
We shouldn't cache things. we should just fetch from `raw` directly. Single source of truth.

Reply:
I'm right in the middle of a bunch of updates to the sensors framework based on your earlier feedback. All properties will now be retrieved directly from the raw instance variable and not cached.